лабораторная3/task1.py

```diff
@@ -0,0 +1,50 @@
+class Book:
+    """ Базовый класс книги. """
+    def __init__(self, name: str, author: str):
+        self._name = name
+        self._author = author
+
+    def __str__(self):
+        return f"Книга {self.name}. Автор {self.author}"
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def author(self):
+        return self._author
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(name={self.name!r}, author={self.author!r})"
+
+
+class PaperBook(Book):
+    def __init__(self, name: str, author: str, pages: int):
+        super().__init__(name, author)
+        if not isinstance(pages, int):
+            raise TypeError("Количество страниц должно быть типа int")
+        if pages <= 0:
+            raise ValueError("Количество страниц должно быть положительным числом")
+        self.pages = pages

```

Comment:
Необходимо в классе PaperBook добавить свойства переменной `pages`, а также проверки перенести в `setter`
```suggestion
        @property
        def pages(self) -> int:
            return self._pages
            
        @pages.setter
        def pages(self, value: int): 
            if not isinstance(value, int):
                raise TypeError("Количество страниц должно быть типа int")
            if value <= 0:
                raise ValueError("Количество страниц должно быть положительным числом")
            self._pages = value
```